src/hooks/api/use-cards-client.ts

```diff
@@ -46,8 +51,26 @@ export function useCardsClient() {
   }
 
   // GET: /Cards​/Deck={deckId}
-  async function getCardsByDeckId(deckId: number): Promise<Card[]> {
-    throw new Error('Not implemented');
+  async function getCardsByDeckId(deckId: string): Promise<Card[]> {
+    const cardsData = await fetchWrapper.get(`/Cards/Deck=${deckId}`);
+
+    const cards: Card[] = cardsData.map((obj: any): Card => {
+      const card = createNewCard();
+      card.id = obj['id'];
+      card.front = {
+        language: obj['flang'],
+        text: obj['ftext'],
+        audio: obj['faud'],
+      };
+      card.back = {
+        language: obj['blang'],
+        text: obj['btext'],
+        audio: obj['baud'],
+      };
+      return card;
+    });
```

Comment:
You are definitely right about needing a method to convert to JSON. If we convert the Deck to a class and add a static method, we wouldn't have to change the syntax of creating and copying decks, I believe.

Reply:
yeah that would definitely make things a lot cleaner